desc/perturbations.py

```diff
@@ -305,7 +308,9 @@ def perturb(
             value = put(  # parameter values below threshold are set to 0
                 value, np.where(np.abs(value) < 10 * np.finfo(value.dtype).eps)[0], 0
             )
-            setattr(eq_new, key, value)
+            # don't set nonexistent profile
+            if not (key == "c_l" or key == "i_l") or value.size:
```

Comment:
my understanding of this logic is hazy so I'll ask for clarification, this is saying if the key is c_l or i_l, then first part will eval to true, then you will negate it, then you check that there actually is a value to change the attr to?

this sort of confuses me, if the key is neither of `c_l` or `i_l`, then wont this evaluate to true? how is this not setting a nonexistent profile?

Reply:
its equivalent to this:
```
if key in {"i_l", "c_l"}:
    if value.size > 0:
        set_value()
else:
    set_value()
```

So basically it always sets the value, unless the key is c_l or i_l and the value has length 0 (corresponding to an empty profile)

Reply:
Rory had the [same question](https://github.com/PlasmaControl/DESC/pull/251#discussion_r967094615). Do you prefer `if key not in ("c_l", "i_l") or value.size`? It's in a loop; not sure whether making a list would cause speed issue.

Reply:
Ah ok I get  it now. Thanks for the clarification, I think it makes sense it just was not obvious from the first look, I always err on the side of too much comments so maybe if adding that helps that would be fine. But no need to make a commit just to change this comment, as the tests take a while to run now